### PR TITLE
fix(sdk): aws queue client does not delete popped messages

### PR DESCRIPTION
**Pull request by flyingImer**
## Checklist

- [X] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [ ] Description explains motivation and solution
- [X] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: libs/wingsdk/src/shared-aws/queue.inflight.ts [hunk 1]
```diff
@@ -54,14 +55,25 @@ export class QueueClient implements IQueueClient {
...
-    const data = await this.client.send(command);
+    const data = await this.client.send(receiveCommand);
     if (!data.Messages) {
       return undefined;
     }
-    return data.Messages[0].Body;
+
+    const message = data.Messages[0];
+
+    if (message.ReceiptHandle) {
```
Comment (Chriscbr) [inline libs/wingsdk/src/shared-aws/queue.inflight.ts:69]: What does it mean if there's no receipt handle on the received message / we can't delete it? (curious if it should be an error of some kind, or if we should log something...)

**Review by Chriscbr — APPROVED** [review]
LGTM

In the future I wonder if we should rework the Queue API to give the user a bit more control -- e.g. `pop` could return you a `Task` object that can be deleted from the queue with `task.complete()`

File: libs/wingsdk/src/shared-aws/queue.inflight.ts [hunk 2]
```diff
@@ -54,14 +55,25 @@ export class QueueClient implements IQueueClient {
   }
 
   public async pop(): Promise<string | undefined> {
```
Comment (tsuf239) [inline libs/wingsdk/src/shared-aws/queue.inflight.ts:57]: Was it tested on aws?

**Review by tsuf239 — APPROVED** [review]
Tests are passing!
image

File: libs/wingsdk/src/shared-aws/queue.inflight.ts (hunk 2 above)
Comment (flyingImer) [reply libs/wingsdk/src/shared-aws/queue.inflight.ts:57]: yes, marked it as WIP to update the sdk test, it's in now

**Comment by flyingImer** [comment]
> LGTM
> 
> In the future I wonder if we should rework the Queue API to give the user a bit more control -- e.g. `pop` could return you a `Task` object that can be deleted from the queue with `task.complete()`

I actually had a couple of POCs on that track, but ended up dropping them. I *think* the current queue API is closer to the classic data structure interface, so I pivoted to match the simulator behavior.

However, I definitely agree with you on the user control story, especially after seeing messages like "queue is not FIFO" in the code base. Queue API might need to re-position itself a bit between classic queue surface and its characteristic of being distributed in the cloud.

File: libs/wingsdk/src/shared-aws/queue.inflight.ts (hunk 1 above)
Comment (flyingImer) [reply libs/wingsdk/src/shared-aws/queue.inflight.ts:72]: Actually, let me log it to start with

**Comment by Chriscbr** [comment]
Ran the queue tests again, all passed 👍

**Review by Chriscbr — APPROVED** [review]

**Pull request merged**